blocks/embed/embed.js

```diff
@@ -186,7 +187,12 @@ export default function decorate(block) {
     wrapper.className = 'embed-placeholder';
     wrapper.innerHTML = '<div class="embed-placeholder-play"><button type="button" title="Play"></button></div>';
     wrapper.prepend(placeholder);
-    wrapper.addEventListener('click', () => {
+    wrapper.addEventListener('click', async () => {
+      await loadConsentManager();
+      await loadAdobeLaunch();
+      if (getEnvType() === 'live') {
+        await loadGTM();
```

Comment:
I feel this may cause GTM script tag to be added multiple times for every button click. Please check.

Reply:
Sure Satya. I suppose we would like to track those clicks before 3 seconds hence placed Consent Manager along GTM scripts to load first. Though the chances of these clicks should be minimal however let me check if I may place a condition there if GTM script is already loaded, then there is no need to await GTM scripts.

Reply:
@sdmcraft  - I suppose as per the changes in the PR https://github.com/sunstar-foundation/sunstar-foundation/pull/53/commits/3bbf83b826851437eda76f2fc36252fd165e4646

This issue will be taken care automatically for rest of the scripts like loadConsentManager, loadAdobeLaunch. After the refactor of loadscript I suppose if it the script tag is there, then it will automatically resolve the promise otherwise it will load it.

But with loadGTM, there is no such check like that be default. It will be better to remove loadGTM from embed.js. Anyway loadGTM will load via its call under delayed.js